create_mode.go

```diff
@@ -0,0 +1,52 @@
+package zk
+
+import "fmt"
+
+const (
+	FlagPersistent                  = 0
+	FlagEphemeral                   = 1
+	FlagSequence                    = 2
+	FlagEphemeralSequential         = 3
+	FlagContainer                   = 4
+	FlagTTL                         = 5
+	FlagPersistentSequentialWithTTL = 6
+)
+
+type createMode struct {
+	flag        int32
+	ephemeral   bool
```

Comment:
nit: for consistency we can prefix all of these boolean fields with `is` - `isEphemeral`, `isSequential`